extensions/binding/org.eclipse.smarthome.binding.hue/src/main/java/org/eclipse/smarthome/binding/hue/internal/FullHueObject.java

```diff
@@ -58,7 +58,7 @@ protected void setType(final String type) {
      *
      * @return model id
      */
-    public String getModelID() {
+    public @Nullable String getModelID() {
         return modelid;
```

Comment:
So, also the local variable `modelid` is `@Nullable`.
In front of adding `@Nullable` annotation, you should also mark the class `@NonNullByDefault` fist.

Reply:
Done.